sqlite features

cabbadie opened the pull request:
- setup electron api in preload
- setup first query for getting worldList
- db file created with the worlds and tags tables
- some npm packages update

valentinEutrope commented on src/client/package.json:
```diff
@@ -11,9 +11,11 @@
     "react-scripts": "^4.0.3",
-    "styled-bootstrap-grid": "^3.1.1",
+    "sqlite3": "^5.0.2",
+    "styled-bootstrap-grid": "^3.1.2",
```
styled-bootstrap-grid on va pouvoir le virer, ça sert à rien je crois x)

valentinEutrope commented on src/client/src/components/WorldList/WorldList.js:
```diff
@@ -1,42 +1,42 @@
+        toast.error(err.message, { id: "errGetWorldList" });
+      }
+    );
+  }
```
Tu n'a pas besoin de faire un if ici, la bonne pratique veut que tu appel ton API en suivant le lifecycle de ton composant, donc en utilisant un useEffect() qui est appelé automatiquement au moment où le composant est monté : https://fr.reactjs.org/docs/hooks-effect.html 

Tu as juste à lui passé worlds en dépendance : 

```js
useEffect(() => {   
   window.api.getWorldList().then(
      (result) => {
        setWorlds(result);
      },
      (err) => {
        toast.error(err.message, { id: "errGetWorldList" });
      }
    );
}, [worlds]);
```

Il faudra également créer un state de loading et d'erreur, mais je m'en occuperai si tu veux

cabbadie replied:
ah ok ! je connaissais pas bien le format de composants comme ça ^^ j'avais vu que les composants en mode objet avec des classes tout ça, je vais modifier ça

valentinEutrope commented on src/client/src/components/WorldList/WorldList.js:
```diff
@@ -1,42 +1,42 @@
   return (
     <>
       <H2>Vos mondes</H2>
+      <Toaster />
```
note pour plus tard : on l’appellera dans le App.js pour l'avoir directement sur toutes les pages

cabbadie replied:
Je peux le faire en modifiant la PR ;)

valentinEutrope commented on src/client/src/components/WorldList/WorldList.js:
```diff
@@ -1,42 +1,42 @@
               <World onClick={() => history.push(`/world/${world.id}`)}>
                 <WorldName>{world.name}</WorldName>
-                <Tags tags={world.tags} />
+                {world.tags ? <Tags tags={world.tags.split(",")} /> : null}
```
world.tags est une string ?

cabbadie replied:
En db, j'ai une table tag et une table qui permet d'attacher un tag à un world. En fait ma query retourne les tags comme une chaîne "<tag>,<tag>,..." pour chaque world.
Sinon je peux faire en sorte que ça retourne un tableau avec un petit traitement quand je récupère le résultat dans l'api?

valentinEutrope replied:
Recevoir un tableau me semble être un format plus approprié, mais les 2 fonctionnent après

valentinEutrope reviewed (changes requested):
Cool pour les nouveautés ! Juste essaye de faire le useEffect pour l'appel api ;)

cabbadie commented on .gitignore:
```diff
@@ -12,7 +12,8 @@
 # production
-/build
+build
+dist
```
Le "/" en premier marchait pas bien, à la limite il fallait "./", mais les nom tout court c'est ok

cabbadie commented on .gitignore:
```diff
@@ -24,3 +25,5 @@
 yarn-debug.log*
 yarn-error.log*
+
+**/*.sqbpro
```
J'utilise une outil, "DB Browser (SQLite)", qui me permet de gérer la db sqlite et me créé des fichiers projets de ce style

cabbadie commented on src/client/electron-preload.js:
```diff
@@ -1,3 +1,40 @@
+        if (err) reject(err);
+        rows.forEach((row) => {
+          row.tags = row.tags.split(",");
+        });
```
Le petit traitement mis au niveau de l'api pour avoir un tableau direct dans le front

cabbadie commented on src/client/package.json:
```diff
@@ -11,9 +11,10 @@
+    "react-hot-toast": "^2.2.0",
     "react-router-dom": "^5.3.0",
     "react-scripts": "^4.0.3",
-    "styled-bootstrap-grid": "^3.1.1",
```
Viré , re-testé sans lui, ça à l'air de rien impacter !

cabbadie commented on src/client/src/App.js:
```diff
@@ -32,6 +33,7 @@ const App = () => {
           </Switch>
         </Router>
       </Providers>
+      <Toaster />
```
Toaster mis au niveau App

valentinEutrope replied:
nickel !

cabbadie committed:
Features :
- setup electron api in preload
- setup first query for getting worldList
- db file created with the worlds and tags tables
- update the gitignore file

cabbadie reviewed (commented):
J'ai refait les choses un peu plus clean, hésite pas à me dire si tu vois quelque chose d'autre ! :p

valentinEutrope commented on src/client/src/components/WorldList/WorldList.js:
```diff
@@ -1,31 +1,30 @@
+        toast.error(err.message, { id: "errGetWorldList" });
+      }
+    );
+  }, [worlds]);
```
```suggestion
  }, [setWorlds]);
```

Je me suis planté dans mon commentaire :
il faut lui passer setWorlds en dépendance, tu n'utilise pas worlds dans le useEffect ^^

valentinEutrope commented on src/client/src/components/WorldList/WorldList.js:
```diff
@@ -1,31 +1,30 @@
+
+  // Set the worlds only once
+  useEffect(() => {
+    window.api.getWorldList().then(
```
Il me semble qu'il y a une raison mais on ne peut pas d'utiliser fetch() ici ?

valentinEutrope reviewed (approved):
Nickel j'approuve ! Mais juste change la dépendance dans le useEffect() , je me suis planté dans mon commentaire ^^
